app/src/main/java/fr/gaulupeau/apps/Poche/ui/ReadArticleActivity.java

```diff
@@ -746,7 +746,13 @@ private String readRawString(int id) throws IOException {
             return sb.toString();
         } finally {
             if(reader != null) {
-                reader.close();
+                try {
+                    reader.close();
+                } catch (IOException e) {
+                    // unrecoverable exception, only log it and move on
+                    Log.w(TAG, "readRawString() failed to close reader\n" + e.toString()
+                            + "\n" + e.getStackTrace().toString());
+                }
```

Comment:
In this case it doesn't really matter whether we have it logged or not. Even if the exception happens, I don't think the warning will tell us anything (we only care whether we could read the resource, not close it). It is a general advice, I don't mind having logging here.
I'd recommend `Log.w(TAG, "readRawString() failed to close reader", e);`. We pass the exception to logger and it handles the formatting for us.